src/components/Vaults.ts

```
@@ -271,10 +271,12 @@ export default class Vaults {
   public getSingleCollateralVault = async (
     account: string,
     vaultSymbol: SingleCollateralVaultSymbol,
-    network: SingleCollateralVaultNetwork,
+    network: Network,
     signer: ethers.Signer
-  ): Promise<SingleCollateralVault> => {
+  ): Promise<SingleCollateralVault | undefined> => {
     this.initialisationCheck();
+    if (network === "ethereum") return;
+
```

Also should throw, not return undefined

---

refactored